File: c++/two_sum.cpp
#include<iostream>
#include<unordered_map>
using namespace std;

int main(){
    vector<int> nums;
    int target=9;
    unordered_map<int,int> hm;
    for(auto i=0;i<nums.size(); i++){
        auto diff=target-nums.at(i);
        if(hm.find(diff)!=hm.end()){
            cout << hm[diff] << "," << i;
            break;
        }
        hm[nums[i]] = i;
    }
}

File: go/two-pointers/_twosum.go
package twopointers

func _twosum(nums []int, target int) []int {
	_p1 := 0
	_p2 := len(nums) - 1
	var sum int
	for _p1 < _p2 {
		sum = nums[_p1] + nums[_p2]
		if sum > target {
			_p2--
		} else if sum < target {
			_p1++
		} else {
			return []int{_p1 + 1, _p2 + 1}
		}
	}
	return nil
}
